src/sentry/migrations/0310_backfill_slack_settings.py

```diff
@@ -0,0 +1,210 @@
+    NotificationSettingTypes.ACTIVE_RELEASE,
+    NotificationSettingTypes.WORKFLOW,
+]
+
+
+# backfill all entities for a given actor that's either a team or a user
+def backfill_one(apps, item_with_actor):
+    NotificationSetting = apps.get_model("sentry", "NotificationSetting")
+    Team = apps.get_model("sentry", "Team")
+    # backfill to the scope of the entity
+    scope_type = (
+        NotificationScopeType.TEAM
+        if isinstance(item_with_actor, Team)
+        else NotificationScopeType.USER
+    )
+    for notification_type in SETTINGS_TO_BACKFILL:
+        # find the setting for that user/teams
+        settings = NotificationSetting.objects.filter(
+            scope_type=scope_type.value,
+            target_id=item_with_actor.actor_id,
+            provider=ExternalProviders.SLACK.value,
+            type=notification_type.value,
+            scope_identifier=item_with_actor.id,
+        )
+        setting = settings.first()
+        # if we have a setting that is not the default we can skip
+        if setting and setting.value != NotificationSettingOptionValues.DEFAULT.value:
+            continue
+
+        # if we have a row it's set the default which we need to change
+        if setting:
+            setting.value = NotificationSettingOptionValues.NEVER.value
+            setting.save()
+        else:
+            # otherwise create the backfill
+            NotificationSetting.objects.create(
+                scope_type=scope_type.value,
+                target_id=item_with_actor.actor_id,
+                provider=ExternalProviders.SLACK.value,
+                type=notification_type.value,
+                scope_identifier=item_with_actor.id,
+                value=NotificationSettingOptionValues.NEVER.value,
+            )
+
+
+# backfill all users who joined after START_DATE_DEFAULT_SLACK_NOTIFICAITON
+def backfill_users(apps):
+    Identity = apps.get_model("sentry", "Identity")
+    IdentityProvider = apps.get_model("sentry", "IdentityProvider")
+
+    for identity_provider in RangeQuerySetWrapperWithProgressBar(IdentityProvider.objects.all()):
+        if identity_provider.type != "slack":
+            continue
+        for identity in Identity.objects.filter(idp=identity_provider).select_related("user"):
+            user = identity.user
+            # skip users who joined after the cutoff date
+            if user.date_joined >= START_DATE_DEFAULT_SLACK_NOTIFICAITON:
+                continue
+
+            backfill_one(apps, user)
+
+
+# backfill all teams regardless of start date
+def backfill_teams(apps):
+    ExternalActor = apps.get_model("sentry", "ExternalActor")
+    Team = apps.get_model("sentry", "Team")
+
+    for external_actor in RangeQuerySetWrapperWithProgressBar(ExternalActor.objects.all()):
+        if external_actor.provider != ExternalProviders.SLACK.value:
+            continue
+
+        team = Team.objects.get(actor_id=external_actor.actor_id)
+        backfill_one(apps, team)
```

Comment:
Not sure how likely this is but maybe 
```
team = Team.objects.filter(actor_id=external_actor.actor_id).first()
if team:
  backfill_one(apps, team)
``` 
instead?

Reply:
@barkbarkimashark Yea, I'll do something like that. Good idea.